File: Iteration2/code/CSE3063GRP12/src/features/notification/NotificationController.java
package features.notification;

import java.io.IOException;
import java.util.ArrayList;
import core.exceptions.UserNotFoundException;
import core.general_providers.SessionController;
import core.general_providers.TerminalManager;
import core.models.abstracts.User;
import core.models.concretes.Notification;
import core.models.concretes.NotificationResponse;
import core.repositories.NotificationRepositories;
import features.main_menu.MenuController;

public class NotificationController {
    private NotificationView notificationView;
    private NotificationRepositories notificationRepositories;
    private Notification notification;
    //read = true olacak
    //notification listesini tek tek yazacak
    //q ya basarsa main menuye dönecek
    public NotificationController(){
        this.notificationView = new NotificationView();
        this.notificationRepositories = new NotificationRepositories();
        this.notification = new Notification();
        handleNotification();
    }


    private void navigateToMenu() {
        new MenuController();
    }


    private String getUserInput() {
        String input = TerminalManager.getInstance().read();
        return input;
    }

    public void handleNotification(){
        User user = SessionController.getInstance().getCurrentUser();
        NotificationResponse notificationResponse;
        try {
            notificationResponse = notificationRepositories.getNotification(user.getUserName());
            ArrayList<Notification> notifications = notificationResponse.getListOfNotification();
            notificationView.showNotificationList(notifications);
            notificationRepositories.updateNotificationRead(user.getUserName());
            notificationRepositories.updateNotificationResponseRead(user.getUserName(),false);
        } catch (IOException | UserNotFoundException e) {
            System.out.println(e.toString());
        }

        notificationView.showQuitMessage();
        String input = getUserInput();
        if (input.equals("q")) {
            navigateToMenu();
        } 
    }

}

File: Iteration3/code/CSE3063GRP12/src/features/notification/NotificationController.py
from core.models.concretes.Notification import Notification
from core.repositories.NotificationRepository import NotificationRepository
from features.notification.NotificationView import NotificationView
from core.general_providers.SessionController import SessionController
from core.exceptions.UserNotFoundException import UserNotFoundException

class NotificationController:
    def __init__(self):
        self.notification_view = NotificationView()
        self.notification_repository = NotificationRepository()
        self.notification = Notification()
        self.handle_notification()

    def navigate_to_menu(self,message):
        if message == "q":
            from features.main_menu.MenuController import MenuController
            MenuController()
        else:
            print("Unexpected input!")
            message = input("Press q to return Main Menu: ")
            self.navigate_to_menu(message)



    def handle_notification(self):
        user = SessionController.getInstance().get_current_user()
        try:
            notification_response = self.notification_repository.get_notification(user.userName)
            if notification_response is None:
                print("You have no notification")
            else:
                notifications = notification_response.listOfNotification
                self.notification_view.show_notification_list(notifications)
                self.notification_repository.update_notification_read(user.userName)
                self.notification_repository.update_notification_response_read(user.userName, False)

        except (IOError, UserNotFoundException) as e:
            print(str(e))

        input_message = input("Press q to return Main Menu: ")
        self.navigate_to_menu(input_message)
